File: app/src/main/java/es/etologic/mahjongscoring2/data/repository/local/daos/CombinationsDao.java
package es.etologic.mahjongscoring2.data.repository.local.daos;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import es.etologic.mahjongscoring2.domain.entities.Combination;

@Dao
public interface CombinationsDao {

    @Insert (onConflict = OnConflictStrategy.REPLACE)
    void bulkInsert(List<Combination> combinations);

    @Query("SELECT * FROM Combinations ORDER BY combinationPoints, combinationName")
    List<Combination> getAllOrderedByPoints();

    @Query("SELECT * FROM Combinations " +
            "WHERE combinationName LIKE :filter " +
            "OR combinationDescription LIKE :filter " +
            "ORDER BY combinationPoints")
    List<Combination> getFilteredCombinations(String filter);
}

File: app/src/main/java/com/etologic/mahjongscoring2/data_source/local_data_source/local/daos/CombinationsDao.kt
package com.etologic.mahjongscoring2.data_source.local_data_source.local.daos

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.etologic.mahjongscoring2.business.model.entities.Combination
import io.reactivex.Single

@Dao
interface CombinationsDao {
    
    @Query("SELECT * FROM Combinations")
    fun getAll(): Single<List<Combination>>
    
    @Insert
    fun bulkInsert(combinations: List<Combination>)
    
    @Query(
        "SELECT * FROM Combinations " +
            "WHERE combinationName LIKE :filter " +
            "OR combinationDescription LIKE :filter " +
            "ORDER BY combinationPoints, combinationName ASC"
    )
    fun getFiltered(filter: String): Single<List<Combination>>
}
